Add support for Sprockets v4 to the DummyApp

kennyadsl opened the pull request:
~~🛑 **BLOCKED BY https://github.com/rails/sprockets-rails/pull/446**~~

**Description**
Sprockets 4 introduced a mandatory configuration manifest at a specific location. We don't have the ability to set it at that specific location in the DummyApp we use for our specs.

~~I've opened https://github.com/rails/sprockets-rails/pull/446 on sprockets-rails which should fix this issue. I'll leave this PR open (working with that branch) waiting to know if that PR will be accepted.~~

**Update:** Now an empty manifest will be written in the expected location inside `spec/dummy` during the dummy app setup.

Closes #3374, closes #3376

~~After merging this we need to revert~~ This also reverts #3378.

**Checklist:**
- [ ] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [ ] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)

kennyadsl committed:
Add a DummyApp sprockets 4 configuration manifest

This is needed from Sprockets, since v4. It contains
all dependencies that needs to be compiled.

Co-Authored-By: Elia Schito <elia@schito.me>

tvdeyen commented on core/lib/spree/testing_support/dummy_app/assets/config/manifest.js:
```diff
@@ -0,0 +1,6 @@
+//= link spree/frontend/all.css
+//
+//= link spree/backend/all.js
+//= link spree/frontend/all.js
```
I dont think we need to put anything into this file. People could get confused by this and think they need to add this to their app after upgrading to Sprockets 4. By they dont since we put all our assets in the `Rails.application.config.asset.precompile` list and Sprockets 4 still respects this.

kennyadsl replied:
Yes, I think you are right. It was a sort of experiment and it should work either way. I think that using a manifest coherent with what is required to really use Solidus works better. Thanks!

tvdeyen commented on core/lib/spree/testing_support/dummy_app.rb:
```diff
@@ -87,6 +87,7 @@ class Application < ::Rails::Application
     config.action_controller.include_all_helpers = false
 
     if config.respond_to?(:assets)
+      config.assets.config_manifest = File.expand_path('dummy_app/assets/config/manifest.js', __dir__)
```
Love your PR :+1:

tvdeyen reviewed (approved):
Thanks. I think we should leave the file blank on purpose. Actually this file should be only necessary if people actually set `Rails.application.assets.precompile += ['manifest.js']`, but this is something to discuss with the Rails team.

ericsaupe reviewed (approved):
This is great!

elia committed:
Lock rack to be less than 3

Removing the dependency on sprockets < 4 rack is now unbounded, but
rack 3 is only supported from Rails 7.1 up.
